### PR TITLE
modulesync 4.0.0

### PR DESCRIPTION
**Pull request by bastelfreak**
modulesync 2.12.0

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -544,9 +544,3 @@ Style/FormatStringToken:
 # are located.
 RSpec/FilePath:
   Enabled: false
-
-RSpec/MessageSpies:
```
Comment (ekohl) [inline .rubocop.yml:548]: For now this should be kept until someone fixes the actual code

File: spec/spec_helper.rb [hunk 2]
```diff
@@ -1,85 +1,19 @@
...
-require 'rspec-puppet-facts'
-require 'bundler'
-include RspecPuppetFacts
-
-if ENV['DEBUG']
-  Puppet::Util::Log.level = :debug
-  Puppet::Util::Log.newdestination(:console)
-end
 
-# Rough conversion of grepping in the puppet source:
```
Comment (ekohl) [inline spec/spec_helper.rb:18]: These custom facts should be kept in. I started to work on support for this in voxpupuli-test but until then, please keep it.

**Review by ekohl — CHANGES_REQUESTED** [review]
Needs a rebase (and probably some squashing of commits).

**Comment by ekohl** [comment]
18.04 suffers from https://github.com/jenkinsci/packaging/pull/198. There's a workaround mentioned.

**Comment by genebean** [comment]
looks like the plugin ownership is an issue here too @ekohl. Do you want me to dig into that?

**Comment by ekohl** [comment]
> looks like the plugin ownership is an issue here too @ekohl. Do you want me to dig into that?

Please do. For me it did work in a libvirt VM on CentOS 7 and couldn't quite figure out why it fails in Docker.

I haven't looked at Ubuntu 18.04 since this PR adds it.

**Comment by genebean** [comment]
Documented in #1002 and working on a fix in #1001

**Comment by genebean** [comment]
@ekohl I see that the plugin stuff not being idempotent is not new (#839) - is this something we have to get fixed before moving forward or can it be a known issue for the moment? I ask because one of the reasons we started poking at this was that the currently released version does not have the right gpg keys for CentOS based servers while the code on master does. I am all for getting this fixed but it is starting to look like a non-trivial thing to deal with.

**Comment by ekohl** [comment]
This looks like the same problems as we have in master so I'm merging it.

**Pull request merged**